explorer/testdata/destructor/destroy_array.carbon

```diff
@@ -0,0 +1,27 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// RUN: %{explorer} %s 2>&1 | \
+// RUN:   %{FileCheck} --match-full-lines --allow-unused-prefixes=false %s
+// RUN: %{explorer} --parser_debug --trace_file=- %s 2>&1 | \
+// RUN:   %{FileCheck} --match-full-lines --allow-unused-prefixes %s
+// AUTOUPDATE: %{explorer} %s
+// CHECK: DESTRUCTOR A 0
+// CHECK: DESTRUCTOR A 1
+// CHECK: result: 1
+
+package ExplorerTest api;
+
+
+class A{
+    destructor[me: Self]{
+        Print("DESTRUCTOR A {0}",me.n);
+    }
+    var n: i32;
+}
+
+fn Main() -> i32 {
+  var a: [A; 2] = ({.n = 0},{.n = 1});
```

Comment:
This looks backwards -- I would think the second element of the tuple should be destructed before the first, if we're sticking to typical unrolling of destruction.

Reply:
Sorry, you are right!